.github/workflows/ci-tests.yml

```
@@ -54,7 +54,7 @@ jobs:
         pip install .
     - name: Run the tests
       run: |
-        tox -e ruleslinting
+        tox -e ruleslinting -- -n 4 test
```

We use `-n 2` for the CI tests.

Wonder should we consolidate on `2` or `4` or have different ones?

@pwildenhain do you remember why you went for `2` initially?

---

If there are 4 it uses 4 if there are 2 it makes little difference. Most people locally will have at least 4

---

Yeah just wonder if we should use `4` for the other sets rather than `2`? Would that be any faster?

---

Let's see if this is any faster: https://github.com/sqlfluff/sqlfluff/pull/2841

It's not really to do with this PR, but curious and think would be better to be consistent, unless a good reason not to be.

---

Ah looks like they are all two-core machines for GitHub Actions: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

And increasing to 4 is not looking any faster to me in #2841, presumably because of that.

Would you mind changing to two for consistency? That way if we want to change, we can find the all with one search of the code.

```suggestion
        tox -e ruleslinting -- -n 2 test
```

---

So I daw you tested and its not faster in Github, however should I switch them to all 4 or all 2. There is still an argument for 4 as its more common for the dev env. 

:)

---

Well this is YML file is only for GitHub actions and not used locally?

Plus my pathetic machine slows to a crawl when using `sqlfluff lint -p 4`.

So my vote is 2,  but could be persuaded to 4 (automatic improvement if/when GitHub improve their compute).

---

If you want to be clever, you can do this in bash:
```
$(python -c 'import multiprocessing; print(multiprocessing.cpu_count())')
```

That returns the number of CPUs on the current machine.

---

I changed it to 2. Its all the same then and thus at least not confusing.